pyccl/tests/test_cclobject.py

```diff
@@ -0,0 +1,212 @@
+import pytest
+import numpy as np
+import pyccl as ccl
+import functools
+
+
+def test_fancy_repr():
+    # Test fancy-repr controls.
+    cosmo1 = ccl.CosmologyVanillaLCDM()
+    cosmo2 = ccl.CosmologyVanillaLCDM()
+
+    ccl.CCLObject._fancy_repr.disable()
+    assert repr(cosmo1) == object.__repr__(cosmo1)
+    assert cosmo1 != cosmo2
+
+    ccl.CCLObject._fancy_repr.enable()
+    assert repr(cosmo1) != object.__repr__(cosmo2)
```

Comment:
Humm, what is this testing exactly? Won't this always be the case, regardless what you've done to fancy_repr?

Reply:
Correct, should have been `cosmo1` instead of `cosmo2`.